daemons/ipa-sam/ipa_sam.c

```diff
@@ -2419,6 +2419,8 @@ static NTSTATUS ipasam_set_trusted_domain(struct pdb_methods *methods,
 	if (entry == NULL || sid == NULL) {
 		smbldap_make_mod(priv2ld(ldap_state), entry, &mods,
 				 LDAP_ATTRIBUTE_UIDNUMBER, IPA_MAGIC_ID_STR);
+		smbldap_make_mod(priv2ld(ldap_state), entry, &mods,
+		                 LDAP_ATTRIBUTE_GIDNUMBER, IPA_MAGIC_ID_STR);
```

Comment:
@flo-renaud I think we need to change this. Autogeneration with DNA plugin wouldn't work here because a generated group ID does not mean the group object actually will exist. Given that TDO is created in the sub-tree that is not handled by the Managed Entries plugin, we should set gidNumber to a value of existing group.

Luckily, we have SMB Default Group for this. This is our fallback POSIX group for cases like this. Please fetch gidNumber of the SMB Default Group and use it instead here.

```
dn: cn=Default SMB Group,cn=groups,cn=accounts,dc=example,dc=com
cn: Default SMB Group
description: Fallback group for primary group RID, do not add users to this group
objectClass: top
objectClass: ipaobject
objectClass: posixgroup
objectClass: ipantgroupattrs
ipaUniqueID: 19ff163e-e60a-11e6-8794-001a4a62eb77
gidNumber: 1536000003
ipaNTSecurityIdentifier: S-1-5-21-570121326-3336757064-1157332047-1003
```